### PR TITLE
ENT-2717: Fix for JWT being double encoded

### PR DESCRIPTION
**Pull request by ziafazal**
Fix for JWT being double encoded.

### CODE REVIEW

File: enterprise_data/filters.py [hunk 1]
```diff
@@ -24,7 +24,8 @@ def update_session_with_enterprise_data(self, request):
         """
         Make cached call to lms to get an enterprise data and update request session.
         """
-        enterprise_client = EnterpriseApiClient(request.auth)
+        jwt = request.auth.decode() if request.auth else None
+        enterprise_client = EnterpriseApiClient(jwt)
```
Comment (muhammad-ammar) [inline enterprise_data/filters.py]: Can you please add comments to explain what we are doing here? Any idea how we are doing this kind of stuff in other services? Most important question is why this issue is happening in first place?

Comment (ziafazal) [reply enterprise_data/filters.py]: Two reasons comes to mind for this happen either python upgrade or bunch of packages upgraded recently.

**Comment by morenol** [comment]
@jmbowman I think that this is related to https://openedx.atlassian.net/browse/BOM-1434

**Review by macdiesel — APPROVED** [review]
seems reasonable but wondering why there are so many non related changes in this PR?

**Comment by ziafazal** [comment]
@morenol  question why you did not release python 2.7 drop changes?

**Comment by morenol** [comment]
@ziafazal I wanted to merge first https://github.com/edx/edx-enterprise-data/pull/187, which ~we can close if we merge this~ is already merged. I also wanted to merge https://github.com/edx/edx-enterprise-data/pull/182

**Comment by morenol** [comment]
I tested this on my devstack. This solves  https://openedx.atlassian.net/browse/BOM-1434 error. So, I think that with a rebase this is ready to get merged.

**Pull request merged**